src/ssl.jl

```diff
@@ -35,13 +35,22 @@ bio_clear_flags(bio::BIO) = bio_set_flags(bio, 0x00)
 function on_bio_stream_read(bio::BIO, out::Ptr{Cchar}, outlen::Cint)
     try
         bio_clear_flags(bio)
-        io = bio_get_data(bio)::TCPSocket
-        n = bytesavailable(io)
-        if n == 0
-            bio_set_read_retry(bio)
-            return Cint(0)
+        io = bio_get_data(bio)
+        if io isa TCPSocket
```

Comment:
maybe comment about why we have these identical branches (I assume it allows us to avoid a dynamic dispatch in the first branch?)

It makes me wonder how big the perf impact of the dynamic dispatch is if we are switching out TCPSocket (e.g. for an identical implementation that was just of type `TCPSocket2`)?